content/ru/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md

```diff
@@ -0,0 +1,343 @@
+
+```
+NAME            READY     STATUS    RESTARTS   AGE
+liveness-exec   1/1       Running   1          1m
+```
+
+## Определение liveness HTTP запроса
+
+Другой вид liveness пробы использует запрос HTTP GET. Ниже представлен файл конфигурации для Pod, который запускает контейнер, основанный на образе `k8s.gcr.io/liveness`.
+
+{{< codenew file="pods/probe/http-liveness.yaml" >}}
+
+В конфигурационном файле вы можете видеть Pod с одним контейнером.
+Поле `periodSeconds` определяет, что kubelet должен производить liveness
+пробы каждые 3 секунды. Поле `initialDelaySeconds` сообщает kubelet'у, что он должен ждать 3 секунды перед проведением первой пробы. Для проведения пробы
+kubelet отправляет запрос HTTP GET на сервер, который запущен в контейнере и слушает порт 8080. Если обработчик пути `/healthz` на сервере
+возвращает код успеха, kubelet рассматривает контейнер как живой и здоровый. Если обработчик возвращает код ошибки, kubelet убивает и перезапускает контейнер.
+
+Любой код, больший или равный 200 и меньший 400, означает успех. Любой другой код интерпретируется как ошибка.
+
+Вы можете посмотреть исходные коды сервера в
+[server.go](https://github.com/kubernetes/kubernetes/blob/{{< param "githubbranch" >}}/test/images/agnhost/liveness/server.go).
+
+В течение первых 10 секунд жизни контейнера обработчик `/healthz`
+возвращает статус 200. После обработчик возвращает статус 500.
+
+```go
+http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+    duration := time.Now().Sub(started)
+    if duration.Seconds() > 10 {
+        w.WriteHeader(500)
+        w.Write([]byte(fmt.Sprintf("error: %v", duration.Seconds())))
+    } else {
+        w.WriteHeader(200)
+        w.Write([]byte("ok"))
+    }
+})
+```
+
+Kubelet начинает выполнять health checks через 3 секунды после старта контейнера.
+Итак, первая пара проверок будет успешна. Но после 10 секунд health
+checks будут завалены и kubelet убьёт и перезапустит контейнер.
+
+Чтобы попробовать HTTP liveness проверку, создайте Pod:
+
+```shell
+kubectl apply -f https://k8s.io/examples/pods/probe/http-liveness.yaml
+```
+
+Через 10 секунд посмотрите события Pod, чтобы проверить, что liveness probes завалилась и container перезапустился:
+
+```shell
+kubectl describe pod liveness-http
+```
+
+В релизах до v1.13 (включая v1.13), если переменная окружения
+`http_proxy` (или `HTTP_PROXY`) определена на node, где запущен Pod,
+HTTP liveness проба использует этот прокси.
+В версиях после v1.13, определение локальной HTTP прокси в переменной окружения не влияет на HTTP liveness пробу.
+
+## Определение TCP liveness пробы
+
+Третий тип liveness проб использует TCP сокет. С этой конфигурацией
+kubelet будет пытаться открыть сокет к вашему контейнеру на определённый порт.
+Если он сможет установить соединение, контейнер будет считаться здоровым, если нет, будет считаться заваленным.
+
+{{< codenew file="pods/probe/tcp-liveness-readiness.yaml" >}}
```

Comment:
I'd expect to see this file copied into the localization unless it's already present.

Reply:
Copy/paste source code? Why not)